src/pages/taskQueue/TaskQueueTable/index.tsx

```diff
@@ -0,0 +1,169 @@
+import { useEffect, useMemo, useRef, useState } from "react";
+import styled from "@emotion/styled";
+import Badge from "@leafygreen-ui/badge";
+import { LGColumnDef, useLeafyGreenTable } from "@leafygreen-ui/table";
+import { Body, Disclaimer } from "@leafygreen-ui/typography";
+import { useTaskQueueAnalytics } from "analytics";
+import { StyledRouterLink, WordBreak } from "components/styles";
+import { BaseTable } from "components/Table/BaseTable";
+import { TablePlaceholder } from "components/Table/TablePlaceholder";
+import {
+  getVersionRoute,
+  getTaskRoute,
+  getUserPatchesRoute,
+} from "constants/routes";
+import { TaskQueueItem, TaskQueueItemType } from "gql/generated/types";
+import { formatZeroIndexForDisplay } from "utils/numbers";
+import { msToDuration } from "utils/string";
+
+type TaskQueueColumnData = Omit<TaskQueueItem, "revision">;
+interface TaskQueueTableProps {
+  taskQueue: TaskQueueColumnData[];
+  taskId?: string;
+}
+const estimateSize = () => 65;
+
+const TaskQueueTable: React.FC<TaskQueueTableProps> = ({
+  taskId,
+  taskQueue = [],
+}) => {
+  const taskQueueAnalytics = useTaskQueueAnalytics();
+  const tableContainerRef = useRef<HTMLDivElement>(null);
+  const [selectedRowIndexes, setSelectedRowIndexes] = useState<number[]>([]);
+  const columns = useMemo(
+    () => taskQueueTableColumns(taskQueueAnalytics.sendEvent),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [],
+  );
+
+  const table = useLeafyGreenTable<TaskQueueColumnData>({
+    data: taskQueue,
+    columns,
+    containerRef: tableContainerRef,
+    useVirtualScrolling: true,
+    enableColumnFilters: false,
+    virtualizerOptions: {
+      estimateSize,
+    },
+  });
+  const performedInitialScroll = useRef(false);
+  useEffect(() => {
+    if (taskId !== undefined && !performedInitialScroll.current) {
+      const i = taskQueue.findIndex((t) => t.id === taskId);
+      setSelectedRowIndexes([i]);
+      table.scrollToIndex(i, { align: "start" });
+
+      setTimeout(() => {
+        performedInitialScroll.current = true;
+        table.scrollToIndex(i, { align: "start" });
```

Comment:
nit. not a big deal but `"center"` could be preferable to prevent the problem where the `start` doesn't seem to properly take into account the sticky header height (but also I don't really mind either way)
<img width="1461" alt="Screenshot 2024-03-04 at 2 12 16 PM" src="https://github.com/evergreen-ci/spruce/assets/47064971/00ca2015-2d81-4dd7-837c-0928bc858d70">